public/app/features/dashboard/services/DashboardLoaderSrv.ts

```diff
@@ -148,41 +148,54 @@ export class DashboardLoaderSrv {
   }
 
   /**
-   * This is a temporary solution to load dashboards dynamically from a datasource
-   * Eventually this should become a plugin type or a special handler in the dashboard
-   * loading code
+   * Load a dashboard from repository
    */
-  async _loadFromDatasource(dsid: string) {
-    const ds = await getDatasourceSrv().get(dsid);
-    if (!ds) {
-      return Promise.reject('can not find datasource: ' + dsid);
-    }
-
+  async _loadFromRepository(repo: string, uid: string): Promise<DashboardDTO> {
     const params = new URLSearchParams(window.location.search);
-    const path = params.get('path');
-    if (!path) {
-      return Promise.reject('expecting path parameter');
-    }
-
-    const queryParams: { [key: string]: string } = {};
+    const ref = params.get('ref'); // commit hash or branch
 
-    params.forEach((value, key) => {
-      queryParams[key] = value;
-    });
-
-    return getBackendSrv()
-      .get(`/api/datasources/uid/${ds.uid}/resources/${path}`, queryParams)
-      .then((data) => {
+    const url = `apis/provisioning.grafana.app/v0alpha1/namespaces/${config.namespace}/repositories/${repo}/files/${uid}`
+    return getBackendSrv().get(url, ref ? {ref} : undefined).then( (v) => {
+      // raw spec
+      const dashboard = v.resource.dryRun.spec as DashboardDataDTO;
+      if (dashboard) {
         return {
           meta: {
-            fromScript: true,
-            canDelete: false,
-            canSave: false,
             canStar: false,
+            isSnapshot: false,
+            canShare: false,
+
+            // Should come from the repo settings
+            canDelete: true,
+            canSave: true,
+            canEdit: true,
+
+            // Indicate that the file came from a repository
+            repository: {
+              url,
+              title: repo,
+            }
           },
-          dashboard: data,
-        };
-      });
+          dashboard,
+        }
+      }
+
+      return Promise.resolve({
+        meta: {
+          canStar: false,
+          isSnapshot: false,
+          canDelete: false,
+          canSave: false,
+          canEdit: false,
+          canShare: false,
+          isDashboardNotFound: true // <<<<<<<<
+        },
+        dashboard: { 
+          title: 'Hello: '+ uid + " // " + ref, 
+          schemaVersion: 0, 
+        },
+      } as any)
```

Comment:
Is `as any` required to make this work? Can we possibly have a typed constant or something to avoid it? 😅